### ISSUE TITLE
Slow particle effects with carpet build 2 and optifine

### ISSUE THREAD

**Issue by Scorpion34** [gnembon/fabric-carpet#393]
Game runs smoothly but the particle effects generated by potion effects or the particles from breaking a block are pretty slow the particles work fine without optifine same goes for only using optifine and not carpet particles work good as well not a major bug but it can get very annoying after awhile

**Comment by Ghoulboy78** [comment]
Turn off particles, or use a mod like tweakeroo to make them less laggy

**Comment by Scorpion34** [comment]
> Turn off particles, or use a mod like tweakeroo to make them less laggy

But its like really hard to explain in text its like your at 30 fps no lag spikes or anything and they still lag and even at 200+ fps they still lag does tweakaroo fix this issue 🤔

**Comment by Ghoulboy78** [comment]
I think so, im not too sure. I think there are some mods that you can ask about in Scicraft discord

**Comment by Scorpion34** [comment]
> I think so, im not too sure. I think there are some mods that you can ask about in Scicraft discord

thank you for the mod suggestion it's very helpful for things I had previously been looking for a solution but the particles are still pretty slow searched around for a fix and the only one is to completely remove particles but I don't really want to do that but anyways with Carpet mod the error GL 1281 pops up which is a mod not in sync now this mod is carpet mod so I think this is the reason some things are slow now I could be wrong but I was curious if you or gnembon could experiment with latest release of optifine, optifabric and carpet mod build 2 you'll see what I'm talking about with particles, splash potions, breaking blocks, dragons breath, lingering potions anything with particles lags even though the game is running smoothly no block lag no fps drops nothing just the particles.

**Comment by Ghoulboy78** [comment]
use latest build then to try pls?

**Comment by Scorpion34** [comment]
> use latest build then to try pls?

yeah I checked every mod I use and updated them slow particles were still present with latest release of optifabric and optifine and carpet mod build 2 the issue still persists it's hard to explain in text form that's why I was curious if you or gnembon could see what I'm talking about with these 3 mods I really just think there's something in carpet mod that is messing with optifine and causing incompatibilities not entirely sure though so if you could try this out for yourself so I don't look like I'm crazy dealing with this bug but I even tried using just carpet mod build 2 optifine pre 10 and optifabric latest release and the issue is still there (of course with the latest version of fabric api as well) really just wanna be acknowledged that I'm not crazy and this is an actual bug with these mods if you don't notice the particle lag early on using a haste 2 beacon and efficiency 5 pickaxe on stone or anything that is instant mineable is how I first noticed this bug.

**Comment by gnembon** [comment]
You could use some '.' in your messages its quite hard to follow on what's going on without punctuation is harder ;)

I will check the particles impact - its possible optifine is here to blame due to the nature of the way they mod the game. I will see if its something I can replicate, or fix, but I wouldn't count on it performing better. For better client side performance, I recommend sodium, not optifine.

**Comment by Scorpion34** [comment]
> You could use some '.' in your messages its quite hard to follow on what's going on without punctuation is harder ;)
> 
> I will check the particles impact - its possible optifine is here to blame due to the nature of the way they mod the game. I will see if its something I can replicate, or fix, but I wouldn't count on it performing better. For better client side performance, I recommend sodium, not optifine.

Yeah sorry if it was hard to understand not very good with punctuation unfortunately.😕 and if it is optifines fault I do apologize for creating an issue that wasn't the cause if I had a lot more mods than I did I'd probably use lithium or sodium or something different from optifine. my main thing with optifine is though is I've used it ever since it was really picked up upon the community and as well as I try to keep my mods as close to vanilla as possible regardless if you do find the issue if I could be informed I'd greatly appreciate it whether it was carpet mod or optifine or even something to do with optifabric considering how its support has been suspended I'd appreciate it :)

**Comment by Scorpion34** [comment]
> You could use some '.' in your messages its quite hard to follow on what's going on without punctuation is harder ;)
> 
> I will check the particles impact - its possible optifine is here to blame due to the nature of the way they mod the game. I will see if its something I can replicate, or fix, but I wouldn't count on it performing better. For better client side performance, I recommend sodium, not optifine.

This issue is still present for the client side with Optifine and Carpet but I found a nice compromise to fix this issue so since carpet mod is a client and server-sided mod I was able to put carpet mod on my server and keep optifine on my client and now both work with 0 issues which is very nice I do still think this issue should be explored though whether it truly is optifine or carpet mods fault either or I'd just like to see this issue get fixed or an explanation behind it

**Comment by LiterallyThis** [comment]
Apologies if I am unnecessarily resurrecting this thread.

I was just experiencing this same issue of slow/choppy particle effects with the latest versions of both Carpet & Optifine installed for 1.16.2. Removing Carpet resolved the issue. I have a high-end machine w/ an RTX 2060, so I am fairly confident that this issue does not stem from my hardware. To further clarify, my FPS was perfectly normal with or without Carpet installed; but particle effects were noticeably choppy.

Let me know if I can provide any more details here, I would love to be able to run Carpet with Optifine.

**Comment by James103** [comment]
Could it be that the particle FPS is somehow locked to the game TPS?

**Comment by Scorpion34** [comment]
> Could it be that the particle FPS is somehow locked to the game TPS?

No even with 20 TPS it lags and MSPT has no effect either
